serialization.rst

```diff
@@ -222,6 +225,53 @@ possible mapping values are:
   if it exists, will be sent as an associative array to the ``bsonUnserialize``
   function to initialise the object's properties.
 
+Scalar Types
+------------
+
+For types that the driver has a wrapper class for, it is possible to define a
+map from each supported data type to a class of your own, providing it
+implements the ``MongoDB\Driver\TypeWrapper`` interface.
+
+The supported data types are Binary, Decimal128, Javascript, MaxKey, MinKey,
+ObjectID, Regex, Timestamp, and UTCDateTime.
+
+The ``MongoDB\BSON\TypeWrapper`` interface defines two functions:
+``fromType()``, which takes a ``MongoDB\BSON\Type`` argument, and
+``toType()``, which returns a ``MongoDB\BSON\Type`` value.
+
+As an example, a wrapped UTCDateTime class, could look like::
+
+    class UTCDateTimeWrapper implements \MongoDB\Driver\TypeWrapper
+    {
+        private $intern;
+
+        function fromType(\MongoDB\BSON\Type $type)
+        {
+            if (! $type instanceof \MongoDB\BSON\UTCDateTime) {
+                throw UnexpectedValueException;
+            }
+
+            $this->intern = $type->toDateTime();
+        }
+
+        function toType() : \MongoDB\BSON\Type
+        {
+            return new \MongoDB\BSON\UTCDateTime( $this->intern );
+        }
+    }
+
+
+If the defined class is inheriting an original ``MongoDB\BSON\*`` class, then
```

Comment:
Our BSON classes are all final, so they wouldn't be inherited. Were you thinking of composition here?